numpy/random/_examples/cython/extending_distributions.pyx

```diff
@@ -72,3 +74,43 @@ def uint10_uniforms(Py_ssize_t n):
     randoms = np.asarray(random_values)
     return randoms
 
+# cython example 3
+def uniforms_ex(bit_generator, Py_ssize_t n, dtype=np.float64):
+    """
+    Create an array of `n` uniformly distributed doubles via a "fill" function.
+
+    A 'real' distribution would want to process the values into
+    some non-uniform distribution
+
+    Parameters
+    ----------
+    bit_generator: BitGenerator instance
+    n: int
+        Output vector length
+    dtype: {str, dtype}, optional
+        Desired dtype, either 'd' (or 'float64') or 'f' (or 'float32'). The
+        default dtype value is 'd'
+    """
+    cdef Py_ssize_t i
+    cdef bitgen_t *rng
+    cdef const char *capsule_name = "BitGenerator"
+    cdef np.ndarray randoms
+
+    _dtype = np.dtype(dtype)
+    if _dtype.type not in (np.float32, np.float64):
+        raise TypeError('Unsupported dtype "%r"' % dtype)
```

Comment:
Could skip this step entirely, and just perform it in the `else` of the `if` used to choose a C function down below.

Reply:
ok